### PR TITLE
ORION-39 show all git credential options during git import

### PR DESCRIPTION
**Pull request by aaudiber**

### CODE REVIEW

**Comment by aaudiber** [comment]
@mingyukim for SA

File: bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js [hunk 1]
```diff
@@ -81,8 +81,8 @@ define(['i18n!orion/nls/messages', 'orion/webui/littlelib'],
 				this._activeElements.onClose = onClose;
 				var focusNode = fillFunction(this._activeElements.parameterArea, this._activeElements.dismissArea);
 				if (!focusNode) {
-					// no parameters were generated.  
-					return false;
+					// no required parameters
+					return true;
```
Comment (mingyukim) [inline bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js:85]: Should you get this information programmatically rather than assuming there's no required params?

File: bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js [hunk 2]
```diff
@@ -328,6 +318,24 @@ define(['i18n!orion/nls/messages', 'orion/webui/littlelib'],
...
+
+					var options = makeButton(messages["More"], parentDismiss); //$NON-NLS-0$
+					var activateMoreOptions = function() {
+						commandInvocation.parameters.optionsRequested = true;
+						finish(self);
+					}
+					if (Object.keys(commandInvocation.parameters.parameterTable).length === 0) {
+						activateMoreOptions();
+					} else {
+						options.addEventListener("click", function() { //$NON-NLS-0$
```
Comment (mingyukim) [inline bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js:333]: activeMoreOptions is already a function, so you should be able to just pass it in, right?

File: bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js [hunk 3]
```diff
@@ -328,6 +318,24 @@ define(['i18n!orion/nls/messages', 'orion/webui/littlelib'],
 					localClose();
 					if (typeof(cancelFunction) === 'function') cancelFunction();
 				}, false);
+
+				if (commandInvocation.parameters.hasOptionalParameters()) {
```
Comment (mingyukim) [inline bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js:322]: why did you move this chunk of code down?

File: bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js (hunk 2 above)
Comment (aaudiber) [inline bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js:333]: done

File: bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js (hunk 3 above)
Comment (aaudiber) [inline bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js:322]: it was strange calling finish(self) and then executing more code to create buttons for the already-finished dialog. The alternative was returning early, but there was nothing sensible to return. Instead I moved the code to the end so that finish(self) is called after everything is set up, and we fall through to the single return point.

File: bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js (hunk 1 above)
Comment (aaudiber) [inline bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js:85]: We would ideally not have to rely on the return value to tell us when there are no required parameters, but unfortunately none of the three parameters to this function, commandNode, fillFunction, or onClose has that information. Changing the method signature makes merge conflicts worse and forces us to update all other places in the codebase where it is used.

Comment (mingyukim) [inline bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js:85]: Can you at least make it clear that this return value should be kept in sync with what's in gitPlugins.js?

File: bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js (hunk 3 above)
Comment (mingyukim) [inline bundles/org.eclipse.orion.client.ui/web/orion/parameterCollectors.js:322]: Ok.

**Comment by aaudiber** [comment]
We're not going to merge this for now because we are planning on using Git's local protocol (file:///) to clone Git repos, and it doesn't require any credentials.
